tools/clang/tools/dxcompiler/dxcpdbutils.cpp

```diff
@@ -958,9 +958,13 @@ struct DxcPdbUtils : public IDxcPdbUtils2
   HRESULT SetEntryPointToDefaultIfEmpty() {
     // Entry point might have been omitted. Set it to main by default.
     // Don't set entry point if this instance is non-debug DXIL and has no arguments at all.
-    // TODO: Check to see that this DxilContainer is not a library before setting the entry point.
     if ((!m_EntryPoint || m_EntryPoint->GetStringLength() == 0) && !m_ArgPairs.empty()) {
-      IFR(Utf8ToBlobWide("main", &m_EntryPoint));
+      // Don't set the name if the target is a lib
+      if (m_TargetProfile->GetStringLength() < 3 ||
```

Comment:
We should probably also check `m_TargetProfile` is nullptr. The code around here is a bit messy and it's not clear what m_TargetProfile would be in every code path. I think it'd be best to do:
```
if (!m_TargetProfile ||
    m_TargetProfile->GetStringLength() < 3 ||
    0 != wcsncmp(m_TargetProfile->GetStringPointer(), L"lib", 3)
```

Reply:
Good point. Will do.